debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisMemoryThreshold.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.redis;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.storage.redis.RedisClient;
+import io.debezium.util.IoUtil;
+import io.smallrye.mutiny.tuples.Tuple2;
+
+public class RedisMemoryThreshold implements Supplier<Boolean> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedisMemoryThreshold.class);
+
+    private static final String INFO_MEMORY = "memory";
+    private static final String INFO_MEMORY_SECTION_MAXMEMORY = "maxmemory";
+    private static final String INFO_MEMORY_SECTION_USEDMEMORY = "used_memory";
+
+    private static final Supplier<Boolean> MEMORY_OK = () -> true;
+
+    private RedisClient client;
+
+    private int memoryThreshold;
+
+    private long memoryLimit;
+
+    private Supplier<Boolean> isMemoryOk;
+
+    public RedisMemoryThreshold(RedisClient client, int memoryThreshold, int memoryLimitMb) {
```

Comment:
I'd pass the full config object, not just two memory settings as it can be obtained form that and would be easier to exand ion in future if additional settings are present.